### PR TITLE
UICIRC-738: UI tests replacement with RTL/Jest for FinePolicy/FinePol…

### PR DESCRIPTION
**Pull request by pavelspichonak**
## Purpose
Add RTL/Jest testing for `FinePolicyDetail` component in `src/settings/FinePolicy`

## Refs
https://issues.folio.org/browse/UICIRC-738

## Screenshots
Screen Shot 2022-01-13 at 5 04 36 PM

### CODE REVIEW

File: src/settings/FinePolicy/FinePolicyDetail.test.js [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+      expect(FinesSection).toHaveBeenLastCalledWith(
+        expect.objectContaining({
+          fineSectionOpen: false,
+        }), {}
+      );
+    });
+  });
+
+  describe('handleSectionToggle method', () => {
+    it('should extapnd accordion', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/FinePolicy/FinePolicyDetail.test.js:190]: typo => expand

File: src/settings/FinePolicy/FinePolicyDetail.test.js [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    it('should render FinesSection component', () => {
+      expect(FinesSection).toHaveBeenCalledWith(expect.objectContaining({
+        policy: mockFinePolicyReturnValue,
+        fineSectionOpen: accordionDefaultStatus.viewFineSection,
+      }), {});
+    });
+  });
+
+  describe('handleExpandAll method', () => {
+    it('should extapnd all accordions', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/FinePolicy/FinePolicyDetail.test.js:134]: typo => expand

File: src/settings/FinePolicy/FinePolicyDetail.test.js [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+      expect(Accordion).toHaveBeenLastCalledWith(
+        expect.objectContaining({
+          open: false,
+        }), {}
+      );
+    });
+  });
+
+  describe('getCheckboxValue method', () => {
+    it('should get checkbox value when seletedValue is true', () => {
```
Comment (Stanislau-Snapok) [inline src/settings/FinePolicy/FinePolicyDetail.test.js:234]: I see that it's not your mistake, but can we change `seletedValue` to `selectedValue` for the whole component and tests?

File: src/settings/FinePolicy/FinePolicyDetail.test.js [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+        <div data-testid="finesSection">
+          {getCheckboxValue(true)}
+        </div>
+      ));
+
+      render(
+        <FinePolicyDetail {...testDefaultProps} />
+      );
+
+      expect(getById('finesSection').getByText(labelIds.finePolicyYes)).toBeVisible();
```
Comment (Stanislau-Snapok) [inline src/settings/FinePolicy/FinePolicyDetail.test.js:245]: I think it's better to move all ids into the one object

File: src/settings/FinePolicy/FinePolicyDetail.js [hunk 5]
```diff
@@ -71,10 +72,13 @@ class FinePolicyDetail extends React.Component {
...
   render() {
     const {
-      initialValues: policy = {},
+      initialValues: policy,
       stripes: {
         connect,
-      }
+      },
+      intl: {
+        formatMessage,
```
Comment (Stanislau-Snapok) [inline src/settings/FinePolicy/FinePolicyDetail.js:80]: Could you please check posibility of `formatMessage` using in all cases and drop `FormattedMessage`?

**Review by Stanislau-Snapok — APPROVED** [review]

File: src/settings/FinePolicy/FinePolicyDetail.test.js (hunk 4 above)
Comment (pavelspichonak) [reply src/settings/FinePolicy/FinePolicyDetail.test.js]: done

File: src/settings/FinePolicy/FinePolicyDetail.js (hunk 5 above)
Comment (pavelspichonak) [reply src/settings/FinePolicy/FinePolicyDetail.js:80]: I'm not sure we need to do it. We use `formatMessage` to be able to test some cases. So if we can test without `formatMessage` I would leave `FormattedMessage` component. Let us discuss

Comment (Stanislau-Snapok) [reply src/settings/FinePolicy/FinePolicyDetail.js:80]: Ok, no problem, your point of view make sense.

File: src/settings/FinePolicy/FinePolicyDetail.test.js [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+          {...testDefaultProps}
+          initialValues={{
+            a: {
+              b: 'testValue',
+            },
+          }}
+        />
+      );
+
+      expect(getById(mockTestIds.overdueAboutSection).getByText('testValue')).toBeVisible();
```
Comment (pavelspichonak) [inline src/settings/FinePolicy/FinePolicyDetail.test.js:288]: move testValue to constant

**Review by Dmitriy-Litvinenko — APPROVED** [review]

File: src/settings/FinePolicy/FinePolicyDetail.test.js (hunk 6 above)
Comment (pavelspichonak) [reply src/settings/FinePolicy/FinePolicyDetail.test.js]: done

**Pull request merged**